core/templates/domain/exploration/read-only-exploration-backend-api.service.spec.ts

```diff
@@ -95,139 +72,150 @@ describe('Read only exploration backend API service', function() {
+    expect(readOnlyExplorationBackendApiService.isCached('0')).toBe(true);
 
     // The exploration should be loadable from the cache.
-    ReadOnlyExplorationBackendApiService.loadLatestExploration('0').then(
+    readOnlyExplorationBackendApiService.loadLatestExploration('0').then(
       successHandler, failHandler);
     expect(successHandler).toHaveBeenCalledWith(sampleDataResults);
     expect(failHandler).not.toHaveBeenCalled();
 
     // Resetting the cache will cause another fetch from the backend.
-    ReadOnlyExplorationBackendApiService.clearExplorationCache();
-    expect(ReadOnlyExplorationBackendApiService.isCached('0')).toBe(false);
+    readOnlyExplorationBackendApiService.clearExplorationCache();
+    expect(readOnlyExplorationBackendApiService.isCached('0')).toBe(false);
 
-    $httpBackend.expect('GET', '/explorehandler/init/0').respond(
-      sampleDataResults);
-    ReadOnlyExplorationBackendApiService.loadLatestExploration('0').then(
+    readOnlyExplorationBackendApiService.loadLatestExploration('0').then(
       successHandler, failHandler);
-    $httpBackend.flush();
+    var req = httpTestingController.expectOne(
+      '/explorehandler/init/0');
+    expect(req.request.method).toEqual('GET');
+    req.flush(sampleDataResults);
 
-    expect(successHandler).toHaveBeenCalledWith(sampleDataResults);
+    flushMicrotasks();
+
+    expect(successHandler).toHaveBeenCalled();
     expect(failHandler).not.toHaveBeenCalled();
-  });
+  }));
 
-  it('should report a cached exploration after caching it', function() {
-    var successHandler = jasmine.createSpy('success');
-    var failHandler = jasmine.createSpy('fail');
+  it('should report a cached exploration after caching it', fakeAsync(() => {
+    let successHandler = jasmine.createSpy('success');
+    let failHandler = jasmine.createSpy('fail');
 
     // The exploration should not currently be cached.
-    expect(ReadOnlyExplorationBackendApiService.isCached('0')).toBe(false);
+    expect(readOnlyExplorationBackendApiService.isCached('0')).toBe(false);
 
     // Cache a exploration.
-    ReadOnlyExplorationBackendApiService.cacheExploration('0', {
+    readOnlyExplorationBackendApiService.cacheExploration('0', {
       id: '0',
       nodes: []
     });
 
     // It should now be cached.
-    expect(ReadOnlyExplorationBackendApiService.isCached('0')).toBe(true);
+    expect(readOnlyExplorationBackendApiService.isCached('0')).toBe(true);
 
     // A new exploration should not have been fetched from the backend. Also,
     // the returned exploration should match the expected exploration object.
-    ReadOnlyExplorationBackendApiService.loadLatestExploration('0').then(
+    readOnlyExplorationBackendApiService.loadLatestExploration('0').then(
       successHandler, failHandler);
 
-    // http://brianmcd.com/2014/03/27/
-    // a-tip-for-angular-unit-tests-with-promises.html
-    $rootScope.$digest();
```

Comment:
Can you please explain why we don't need this anymore?

Reply:
$rootScope.$digest is AngularJS native thing to run the digest cycle. We have flushmicrotasks to flush all the requests in Angular.

Reply:
Ah okay. Thanks for explaining @bansalnitish!